src/xbot/common/properties/DatabaseStorageBase.java

```diff
@@ -0,0 +1,175 @@
+package xbot.common.properties;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author John
+ */
+public abstract class DatabaseStorageBase extends PermanentStorageProxy {
+
+    private static Logger log = Logger.getLogger(DatabaseStorageBase.class);
+    
+    private String dbUrlPreFormat = "jdbc:derby:%1s;create=true";
+    private String dbUrl = "";
+    
+    public DatabaseStorageBase(String databaseDirectory) {
+        super();
+        
+        dbUrl = String.format(dbUrlPreFormat, databaseDirectory);
+    }
+    
+    private boolean propertiesTableExists(Connection conn)
+    {
+		try {
+			DatabaseMetaData md = conn.getMetaData();
+            ResultSet tables = md.getTables(null, null, "PROPERTIES", null);
+            
+            if (tables.next())
+            {
+            	return true;
+            }
+            return false;
+	        
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
```

Comment:
We should replace the code in here with something such as logging, and remove the TODO comment.